packages/foam-vscode/static/dataviz/graph.js

```diff
@@ -63,6 +64,7 @@ let model = {
    * in the case it fails, use the fallback style values.
    */
   style: defaultStyle,
+  view: "default",
```

Comment:
Also, there is something that I am a bit concerned about, let me know your thoughts:

Imagine the scenario of having a VS Code command that resets the filtering on the notes in the graph.
DAT uses 2 way data binding, which means that in the current approach to achieve this I would have to change the model of DAT to achieve the result. This feels wrong to me.

Instead we should have a `showNodesByType` which has an array of types, or `null/undefined/[]` if we want to show everything.
This means this would need to be kept in sync with the DAT gui.
Crucially, the source of truth need to be the core model, not the DAT gui model.

I have seen two things that might help with this:
- in https://github.com/vasturiano/force-graph there is a `onRenderFramePost` that we could use to sync the state of the DAT gui with the one of the core model (the reason why I would put it there is because the `render` is already managed by the graph object, we might as well put everything there)
- in https://github.com/dataarts/dat.gui there is an `updateDisplay` function that could do just that. I wonder though if calling it on every render is going to cause issues

Not sure what the ideal here is, maybe this approach would also allow us to keep the state of the DAT gui completely outside of the model (which might be a good thing, the dependency should be `core model => DAT model`, never the other way around, and this would be a good way to enforce it).

What are your thoughts on this?

Reply:
Okay this sounds good. I mostly just work with data so before I started this I read up on MVC, but some things were confusing me. This definitely makes more sense to me. I thought we should have only one model.

Reply:
The pattern I am using for the graph is closer a mix of redux (https://redux.js.org/), SAM (https://sam.js.org/), and meiosis (https://meiosis.js.org/) - in general though this is a unidirectional data flow architecture (with some caveat due to the API of the graph library), as opposed to an MVC architecture.
Let me know if you have any question about this, happy to assist

Reply:
Great. I honestly didn't know where to start. I thought I was on the right track, but I'll check those out.